openapi.yaml

```diff
@@ -47,33 +47,45 @@ info:
+    | 3.  The user logs into the authorization server with their username and password. | 3.  The user logs into the authorization server with their username and password. |
+    | 4.  The authorization server redirects the user to the specificed redirect URL with a temporary authorization `code` (exchange code) in the URL. | 4.  The authorization server redirects the user back to your applaction with an OAuth token in the URL hash. This is temporary and expires in 2 hours without a `refresh_token`. |
+    | 5.  The application issues a POST request (*see below*) to the authentication server with the exchange code, `client_id`, and the client application's `client_secret`. In return, the server returns an OAuth token and `refresh_token`. |  |
+    | 6.  The authentication server responds to the client application with a new OAuth `access_token`. `access_token` is set to expire in 2 hours if not refreshed. |  |
+    | 7.  The refresh token can be used to get a new OAuth `access_token` for another 2 hours by contacting the authentication server again with the `client_id`, `client_secret`, and `refresh_token`. | |
+
+      Once the User has logged into Linode and you have received an exchange code,
+      you will need to trade that exchange code for an Authorization token. You
+      do this by making an HTTP POST request to the following address:
+
+      ```
+      https://login.linode.com/oauth/token
+      ```
 
-    Make this request as `application/x-www-form-urlencoded` or as
-    `multipart/form-data` and include the following parameters in the POST body:
+      Make this request as `application/x-www-form-urlencoded` or as
+      `multipart/form-data` and include the following parameters in the POST body:
 
-    | PARAMETER | DESCRIPTION |
-    |-----------|-------------|
-    | client_id | Your app's client ID |
-    | client_secret | Your app's client secret |
-    | code | The code you just received from the redirect |
+      | PARAMETER | DESCRIPTION | FIND OR UPDATE THIS INFORMATION |
+      |-----------|-------------|---------------------------------|
+      | client_id | Your app's client ID. | [List your OAuth Clients](https://developers.linode.com/api/v4/account-oauth-clients). |
+      | client_secret | Your app's client secret. | [View your OAuth Client](https://developers.linode.com/api/v4/account-oauth-clients-client-id). [Reset Your OAuth Client Secret](https://developers.linode.com/api/v4/account-oauth-clients-client-id-reset-secret/#post). |
+      | code | The code you just received from the redirect | [View your OAuth Client](https://developers.linode.com/api/v4/account-oauth-clients-client-id). [Reset Your OAuth Client Secret](https://developers.linode.com/api/v4/account-oauth-clients-client-id-reset-secret/#post). |
```

Comment:
These two links don't apply to the exchange code, and there's no way to retrieve it if you don't capture it in the redirect.

Reply:
removed the link column because it was confusing